lib/test-requirements-with-tensorflow.txt

```diff
@@ -2,7 +2,6 @@
 # we're not going to update its associated tests anymore. Please don't modify
 # these pinned versions!
 
-keras<2.5.0
 tensorflow>=2.6.0, <2.8.0
```

Comment:
`keras<2.5.0` is resolved to `keras==2.4.3` and  `tensorflow==2.6.0` requires `keras~=2.6` so this requirement is invalid and always causes conflicts. We should add support for new versions of libraries, but this is only used by the `st.cache` decorator, which is deprecated and we have no plans to update it.

Previously this worked because we used the legacy resolver in pip, but it's not a good practice to use this as it may break other dependencies as well.
In other words: The command `pip install 'keras<2.5.0' 'tensorflow>=2.6.0, <2.8.0'` always fails now.

Reply:
We should get rid of the old cache decorator, point users to the new one instead and fix our dependencies 👍